psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,63 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import TYPE_CHECKING
+
+from psycopg.adapt import Dumper, Loader
+from psycopg.pq import Format
+from psycopg.types import TypeInfo
+
+if TYPE_CHECKING:
+    from psycopg.connection import Connection
+    from psycopg.rows import Row
+
+
+def register_shapely_adapters(conn: Connection[Row]) -> None:
+    """Register Shapely dumper and loaders.
+
+    After invoking this function on a connection, the queries retrieving
+    PostGIS geometry objects will return Shapely's shape object instances
+    in text and binary mode.
+
+    Similarly, shape onjects can be sent to the database.
+
+    :param conn: the connection on which to set up the loader and dumper
+
+    """
+    import shapely.wkb
+    from shapely.geometry.base import BaseGeometry
+
+    class GeometryBinaryLoader(Loader):
+        format = Format.BINARY
+
+        def load(self, data: bytes) -> BaseGeometry:
+            return shapely.wkb.loads(data)
+
+    class GeometryTextLoader(Loader):
+        format = Format.TEXT
+
+        def load(self, data: bytes) -> BaseGeometry:
+            # it's a hex string in binary
+            return shapely.wkb.loads(data.decode(), hex=True)
+
+    class GeometryBinaryDumper(Dumper):
```

Comment:
Once we know the `geometry` oid, we should create subclasses which set the `oid` class attribute of the dumpers. You can use `type()` to dynamically create a class. For example:
https://github.com/psycopg/psycopg/blob/2c991429cc28ca50b5ad292f5e42ad5a85b4cf46/psycopg/psycopg/types/hstore.py#L117-L120